addons/account/models/account_move.py

```diff
@@ -3498,6 +3498,10 @@ def _post(self, soft=True):
                 customer_count[invoice.partner_id] += 1
             elif invoice.is_purchase_document():
                 supplier_count[invoice.partner_id] += 1
+            elif invoice.move_type == 'entry':
+                sale_amls = invoice.line_ids.filtered(lambda line: line.balance != 0 and bool(line.partner_id) and line.account_id.account_type == 'asset_receivable')
+                for aml in sale_amls:
+                    customer_count[aml.partner_id] += 1
```

Comment:
I'd suggest doing something like this to avoid counting a partner multiple times if there are several AMLs referencing them.

```
from odoo.tools import groupby
for partner, amls in groupby(sale_amls, lambda line: line.partner_id):
    customer_count[partner] += 1
```